extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -260,18 +247,38 @@ func RestoreRKE1Snapshot(client *rancher.Client, clusterName string, snapshotRes
 }
 
 // RestoreRKE2K3SSnapshot is a helper function to restore a snapshot on an RKE2 or k3s cluster. Returns error if any.
-func RestoreRKE2K3SSnapshot(client *rancher.Client, clusterName string, snapshotRestore *rkev1.ETCDSnapshotRestore, initialControlPlaneValue, initialWorkerValue string) error {
-	clusterObject, existingSteveAPIObject, err := clusters.GetProvisioningClusterByName(client, clusterName, fleetNamespace)
+func RestoreRKE2K3SSnapshot(client *rancher.Client, clusterName string, snapshotRestore *rkev1.ETCDSnapshotRestore, restoreSnapshotConfig *apisV1.Cluster) error {
+	_, existingSteveAPIObject, err := clusters.GetProvisioningClusterByName(client, clusterName, fleetNamespace)
 	if err != nil {
 		return err
 	}
 
-	clusterObject.Spec.RKEConfig.ETCDSnapshotRestore = snapshotRestore
-	clusterObject.Spec.RKEConfig.UpgradeStrategy.ControlPlaneConcurrency = initialControlPlaneValue
-	clusterObject.Spec.RKEConfig.UpgradeStrategy.WorkerConcurrency = initialWorkerValue
+	restoreSnapshotConfig.ObjectMeta.ResourceVersion = existingSteveAPIObject.ObjectMeta.ResourceVersion
 
 	logrus.Infof("Restoring snapshot: %v", snapshotRestore.Name)
-	_, err = client.Steve.SteveType(ProvisioningSteveResouceType).Update(existingSteveAPIObject, clusterObject)
+	cluster, err := client.Steve.SteveType(ProvisioningSteveResouceType).Update(existingSteveAPIObject, restoreSnapshotConfig)
+	if err != nil {
+		return err
+	}
+
+	err = kwait.PollUntilContextTimeout(context.TODO(), 500*time.Millisecond, defaults.ThirtyMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
assuming this was found by observation but confirming 30 minutes is expected timeout

Reply:
Correct. RKE1 can take a painfully long time to finish. This is because the worker node will go into an `Upgrading` state with no real sense of when it will finish. If you have it at 10 or 15 minutes, I've had consistent failures. 30 is excessive but it at least will work each time, I have found 🤷🏿